src/charm.py

```diff
@@ -640,65 +649,95 @@ def _configure_bessd_workload(self) -> None:
             logger.info("Service `routectl` restarted")
             self._bessd_container.restart(self._bessd_service_name)
             logger.info("Service `bessd` restarted")
+        self._wait_for_bessd_grpc_service_to_be_ready(timeout=60)
         self._run_bess_configuration()
 
     def _run_bess_configuration(self) -> None:
         """Runs bessd configuration in workload."""
+        if self._is_bessd_configured():
+            return
+
+        logger.info("Starting configuration of the `bessd` service")
+        command = "/opt/bess/bessctl/bessctl run /opt/bess/bessctl/conf/up4"
+        try:
+            (stdout, stderr) = self._exec_command_in_bessd_workload(
+                command=command,
+                environment=self._bessd_environment_variables,
+                timeout=30,
+            )
+
+            logger.info("Service `bessd` configuration script complete")
+            for line in stdout.splitlines():
+                logger.debug("`up4.bess`: %s", line)
+            if isinstance(stderr, str) and len(stderr) > 0:
```

Comment:
The return value of `_exec_command_in_bessd_workload` is a tuple of strings, so `stderr` should always be a string. This line could be simplified to this:
```suggestion
            if stderr:
```

Reply:
One of the linters was complaining about it in the k8s charm, and I had to use that syntax there. I'll try it to see if the signature is different here and will pass

Reply:
Ok, if this suggestion does not work let me know, this is not a big concern.